python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/search.py

```diff
@@ -127,6 +133,41 @@ def search_facets(
         response = response["aggregateAcrossEntities"]
         return self._parse_facets(response.get("facets", []))
 
+    def list_data_product_assets(
+        self, urn: str, count: int, start: int = 0
+    ) -> SearchResponse:
+        """
+        returns a SearchResponse containing all assets in given data product (by urn)
+        """
+        variables = {
+            "urn": urn,
+            "query": "*",
+            "start": start,
+            "count": count,
+        }
+        try:
+            response = self.graph.execute_graphql(
+                self.data_product_asset_query, variables
+            )
+        except GraphError as e:
+            raise Exception("Unable to execute listDataProductAssets query") from e
```

Comment:
Just a thought - do you think it's worth introducing a custom exception type for these errors?

They can be thrown any time the catalogue is unavailable

Reply:
Yeah quite probably, but another separate ticket